proto/streamlit/proto/NumberInput.proto

```diff
@@ -35,12 +35,12 @@ message NumberInput {
   bool has_max = 12;
 
   DataType data_type = 13;
-  double default = 14;
+  optional double default = 14;
   double step = 15;
   double min = 16;
   double max = 17;
   string help = 18;
-  double value = 19;
+  optional double value = 19;
```

Comment:
Since we're making changes to the proto (and type of the widget value), we should update the proto wrapper class in `element_tree.py`.

Reply:
Whoops, I didn't think about this aspect. I did a quick update to `element_tree`, but I'm not 100% sure how best to integrate this aspect into `element_tree` since `_value=None` already has a slightly different meaning. Do you have a recommendation?

Reply:
I added a default InitialValue class for this. Btw. I need to use isinstance here, otherwise mypy doesn't work correctly with the type checks.

Reply:
I added a default InitialValue class for this. Btw. I need to use isinstance here, otherwise mypy doesn't work correctly with the type checks.